src/prefect/_internal/compatibility/async_dispatch.py

```diff
@@ -19,6 +22,14 @@ def is_in_async_context() -> bool:
         return False
 
 
+def _is_acceptable_callable(obj: Union[Callable, Task]) -> bool:
+    if inspect.iscoroutinefunction(obj):
+        return True
+    if isinstance(obj, Task) and inspect.iscoroutinefunction(obj.fn):
+        return True
+    return False
```

Comment:
Looks like we'll need to bump the version of `prefect` in `prefect-aws` to get this change?